### ISSUE TITLE
ccache: Test Fails, Installation Impossible

### ISSUE THREAD

**Issue by vale981** [NixOS/nixpkgs#37091]
## Issue description
Hi, 
maybe I am handling somehting wrong, but when I try to install ccache, it errors with:
```
test/main
PASSED: 479 assertions, 94 tests, 10 suites
CC='gcc' ./test.sh
./test.sh: line 3516: which: command not found
Compiler:         gcc ()
Compiler version: gcc (GCC) 7.3.0

Running test suite base......................................................
Running test suite nocpp2......................................................
Skipped test suite multi_arch [multiple -arch options not supported on Linux]
Skipped test suite serialize_diagnostics [--serialize-diagnostics not supported by compiler]
Running test suite debug_prefix_map..
FAILED

Test suite:     debug_prefix_map
Test case:      Multiple -fdebug-prefix-map
Failure reason: Relocation (name) not found in test.o

ccache -s:
cache directory                     /tmp/nix-build-ccache-3.3.5.drv-0/ccache-3.3.5/testdir.7115/.ccache
primary config                      /tmp/nix-build-ccache-3.3.5.drv-0/ccache-3.3.5/testdir.7115/ccache.conf
secondary config      (readonly)
cache hit (direct)                     0
cache hit (preprocessed)               0
cache miss                             1
cache hit rate                      0.00 %
cleanups performed                     0
files in cache                         2
cache size                           8.2 kB
max cache size                       5.0 GB

Test data and log file have been left in testdir.7115
make: *** [Makefile:104: test] Error 1
builder for ‘/nix/store/9a0agkaq7amqjvq6ay3bx4bdpnn0xs5c-ccache-3.3.5.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/dnj9nc8v458y0qswsy9bdsf2rbbwg55d-nix-ccache.pl.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/2sp4gz8gdff0770x5zcgf559hf83kcyx-nixos-system-nixos--18.03pre129076.831ef4756e3.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/2sp4gz8gdff0770x5zcgf559hf83kcyx-nixos-system-nixos--18.03pre129076.831ef4756e3.drv’ failed

```

It would be very helpful, if this issue can be resolved.
Thanks a lot. 

### Steps to reproduce
Just add: ``
programs.ccache.enable = true;
`` to your `configuration.nix`

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.15.0, NixOS, 18.03pre129076.831ef4756e3 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.16`
 - channels(root): `"nixos-18.03pre129076.831ef4756e3"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by pbogdan** [comment]
If you can it might be worth updating your system. A PR - https://github.com/NixOS/nixpkgs/pull/36110 - was merged a while ago with a ccache update and build fixes.

**Comment by vale981** [comment]
Thanks.
As my system is quite recent (I set it up the other week), I think that isn't the problem.


On 15 March 2018 21:51:03 CET, pbogdan <notifications@github.com> wrote:
>If you can it might be worth updating your system. A PR -
>https://github.com/NixOS/nixpkgs/pull/36110 - was merged a while ago
>with a ccache update and build fixes.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/NixOS/nixpkgs/issues/37091#issuecomment-373518618

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by pbogdan** [comment]
I may be misreading but the commit your system is on is ~ 3 week old and doesn't include ccache update / fixes, and the logs provided show ccache 3.3.5 being built when both master and nixos-unstable are on 3.4.1?

**Comment by vale981** [comment]
I thank you very much.
Due to my  lack of experience, I thought that I was on unstable. Well I wasn't. The update did the trick.
Thanks :).
